contents/swift/welcome-to-swift.md

```diff
@@ -13,46 +13,52 @@
+Introduced in 2014, Swift has concise and expressive syntax, especially when compared with its predecessor language **Objective-C**. On top of that, Swift also supports playgrounds, a feature that allows programmers to experiment with Swift code and see the results immediately (without the overhead of building and running an app).
 
-Reading the [Language Guide in the official documentation](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/) is definitely recommended, since it explains everything quite clearly, albeit quite verbose.
+## Why should I learn Swift?
 
-If you're just looking to take a dip into Swift, here are the [basics](https://guides.codepath.com/ios/Swift-Basics) you'll need to understand.
+1. **Learning Swift can help you land your dream career.**
+Swift is mainly used for iOS development and iOS developers are well paid. The [salary statistics](https://www.indeed.com/salaries/iOS-Developer-Salaries) shows that iOS developer in US can earn more than $120K per year. In addition, mobile application development is also becoming more popular as of recent times. A [survey in StackOverFlow](https://insights.stackoverflow.com/survey/2017#developer-profile-developer-type) shows that mobile development, although lesser than web development, still takes up a large share of the market.
 
-And if you are really pressed for time, here's a couple of cheatsheets with code examples:
+1. **Swift is easy to learn.** Swift is designed to be a simple and user-friendly language by Apple. Its syntax resembles English, making it easy to read and understand. Any programmer who is familiar with Python, Java or C++ can learn Swift quickly! Even if you are not familiar with any language, Swift is an excellent language to start out with.
 
-- [Swift Cheat Sheet](https://mhm5000.gitbooks.io/swift-cheat-sheet/content/index.html) (web-friendly)
-- [Swift 3.0 Cheat Sheet and Quick Reference](https://koenig-media.raywenderlich.com/uploads/2014/06/RW-Swift-Cheatsheet-0_7.pdf) (print-friendly)
+1. **Swift is an [open source project](https://github.com/apple/swift).** Learning about Swift and contributing to the language may be a useful experience for those interested. Apart from learning about Swift from its documentation, you can also dig into the source code to get new insights. Typically, you can start submitting issues or patches if you find there is any way to improve the language!
+
+Also note that Swift is not perfect. It is still very young, and therefore the language itself may have issues to be addressed, such as [stored properties in extension](https://medium.com/@valv0/computed-properties-and-extensions-a-pure-swift-approach-64733768112c).
+In addition, Swift's libraries/tools are still limited and resources dedicated to earlier Swift versions may not be usable with newer releases.
+However, such limitations also mean that there are plenty of opportunities for you to improve it!
```

Comment:
Let's tighten up this section. There are many things that can be removed without losing value. Also, easy-to-learn, open-source are not really good reasons for learning a language. 

How about this?

---
## Swift Overview

**Swift is the main programming language used for iOS programming.** Introduced in 2014 by Apple, Swift has more concise and more expressive syntax compared to its predecessor language [Objective-C](). Unlike most other software by Apple, Swift is [open source](https://github.com/apple/swift).

**One main attraction to learn Swift's is that iOS developers are well paid** ([example](https://www.indeed.com/salaries/iOS-Developer-Salaries)).

Swift syntax is not too different from other mainstream languages such as Python, Java or C++, which means switching to Swift is not difficult. On top of that, Swift also supports [_playgrounds_](), a feature that allows programmers to experiment with Swift code and see the results immediately, without the overhead of building and running an app.

Here is HelloWorld in Swift:

{give code here}

## Noteworthy Swift Features

Here are some noteworthy Swift features for you to get a feel of Swift.

{I don't think want call these benefits; let's go back to listing features}